rules/options.go

```diff
@@ -66,17 +69,18 @@ type engineOptions struct {
 	lockCoolOff            time.Duration
 	useSharedLockSession   bool
 	useTryLock             bool
+	watchDelay             jitter.DurationGenerator
 }
 
 func makeEngineOptions(options ...EngineOption) engineOptions {
 	opts := engineOptions{
 		concurrency:            5,
 		constraints:            map[string]constraint{},
 		contextProvider:        defaultContextProvider,
-		syncDelay:              1,
+		syncDelay:              jitter.NewDurationGenerator(10*time.Second, syncJitterPercent),
```

Comment:
This is a slight bump to the default delay. This would take it from `1ms` to `10s±10%`, or `between 9s and 11s`. This one might require a smaller delay and higher variance to have the desired effect (few concurrent requests), though I'll defer to your judgement there.